packages/react-app/src/views/Staking.jsx

```diff
@@ -0,0 +1,118 @@
+import { useContractReader } from "eth-hooks"
+import { Button, Input } from "antd"
+import React, { useState } from "react"
+import { ethers } from "ethers"
+import { Link } from "react-router-dom"
+
+/**
+ * web3 props can be passed from '../App.jsx' into your local view component for use
+ * @param {*} yourLocalBalance balance on current network
+ * @param {*} readContracts contracts from current chain already pre-loaded using ethers contract module. More here https://docs.ethers.io/v5/api/contract/contract/
+ * @returns react component
+ **/
+function Staking({ yourLocalBalance, readContracts }) {
+    // you can also use hooks locally in your component of choice
+    // in this case, let's keep track of 'purpose' variable from our contract
+    const purpose = useContractReader(readContracts, "YourContract", "purpose")
```

Comment:
You can delete this, because it's from the example. `"purpose"` is the variable in the contract (`YourContract`), and this line is for monitoring the state of this var.

In the future it may be necessary to keep an eye on the balance staked variable. btw, I suggest using unidirectional data flow throughout the project

Reply:
thanks, deleted.